x-pack/plugins/security_solution/server/lib/detection_engine/signals/single_search_after.ts

```diff
@@ -113,8 +114,32 @@ export const singleSearchAfter = async ({
         searchDuration: '-1.0',
         searchErrors: exc.message,
       };
+    } else if (exc.statusCode === 400) {
+      // added this code to handle https://github.com/elastic/kibana/issues/93333
+      try {
+        const rootCause = JSON.parse(exc.response).error;
+        if (rootCause != null) {
+          let causedByObject = rootCause.caused_by;
+          let reason;
+          while (causedByObject != null) {
+            reason = causedByObject.reason;
+            causedByObject = causedByObject.caused_by;
+          }
+          if (
+            reason != null &&
+            reason.includes(
+              'The nested depth of the query exceeds the maximum nested depth for bool queries set in [indices.query.bool.max_nested_depth]'
+            )
+          ) {
+            throw new Error(
```

Comment:
This `throw` is caught by the `try...catch` here so this error is only logged and then the exception that gets thrown is the original un-parsed exception on line 143

Reply:
Yuuuppp that did it thank you for the 👀  I'm going to wrap the try catch around the `json.parse`